tensorboard/components/tf_backend/router.ts

```diff
@@ -34,7 +36,10 @@ namespace tf_backend {
    *
    * @param dataDir {string=} The base prefix for data endpoints.
    */
-  export function createRouter(dataDir = 'data'): Router {
+  export function createRouter(
+    dataDir = 'data',
+    windowLocationUrl = new URL(window.location.href)
```

Comment:
This is a little scary; in the past, we’ve had issues with code like
this breaking `--path_prefix` handling. It’s fine here because you’re
only using it for the search params rather than the path, but I wonder
if we can make that clearer—something like

```typescript
  export function createRouter(
    dataDir = 'data',
    searchParams = new URLSearchParams(window.location.search)
  ): Router {
    // ...
  }
```

such that future changes to the function don’t accidentally rely on
`path`?